### PR TITLE
adding audio-to-audio widget.

### PR DESCRIPTION
**Pull request by Narsil**
To be done 

After #87 
After #93 
Before #94

### CODE REVIEW

**Review by osanseviero — COMMENTED** [review]
From a quick look it seems fine. Could you please sync now that the widget open sourcing is merged?

I'll take a final look afterwards

**Comment by osanseviero** [comment]
@beurkinger probably interested in this :)

File: widgets/src/lib/InferenceWidget/InferenceWidget.svelte [hunk 1]
```diff
@@ -20,7 +21,8 @@
 
 	export let apiToken: WidgetProps["apiToken"];
 	export let callApiOnMount = false;
-	export let apiUrl = "https://api-inference.huggingface.co";
+	//export let apiUrl = "https://api-inference.huggingface.co";
```
[the remark this answers is not in the document]
Comment (Narsil) [reply widgets/src/lib/InferenceWidget/InferenceWidget.svelte]: Done

File: interfaces/Types.ts [hunk 2]
```diff
@@ -47,6 +48,7 @@ export const PIPELINE_TYPE_PRETTY_NAMES: { [key in PipelineType]: string } = {
 	/// audio
 	"text-to-speech":                                           "Text-to-Speech",
 	"automatic-speech-recognition":                             "Automatic Speech Recognition",
+	"audio-to-audio":                                  			"Audio to Audio(s)",
```
[the remark this answers is not in the document]
Comment (Narsil) [reply interfaces/Types.ts]: Done.

File: widgets/src/lib/InferenceWidget/InferenceWidget.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (Narsil) [reply widgets/src/lib/InferenceWidget/InferenceWidget.svelte]: Done.

**Comment by Narsil** [comment]
Ping @osanseviero (was a simple rebase + cleanup, should be much easier to review).

**Review by osanseviero — APPROVED** [review]
Looks awesome, thanks a lot!
Note that you will need to add this to moon-landing as well, they don't automatically sync at the moment.

File: interfaces/Types.ts [hunk 3]
```diff
@@ -47,6 +48,7 @@ export const PIPELINE_TYPE_PRETTY_NAMES: { [key in PipelineType]: string } = {
 	/// audio
 	"text-to-speech":                                           "Text-to-Speech",
 	"automatic-speech-recognition":                             "Automatic Speech Recognition",
+	"audio-to-audio":                                           "Audio to Audio(s)",
```
Comment (julien-c) [inline interfaces/Types.ts:51]:
```suggestion
	"audio-to-audio":                                           "Audio-to-Audio",
```

looks better like that, no?

File: widgets/src/lib/InferenceWidget/widgets/AudioToAudioWidget/AudioToAudioWidget.svelte [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+				estimatedTime: res.estimatedTime,
+			};
+			getOutput(true);
+		} else if (res.status === "error") {
+			error = res.error;
+		}
+	}
+
+	function parseOutput(body: AudioItem[]): AudioItem[] {
+		for (let item of body) {
```
Comment (julien-c) [inline widgets/src/lib/InferenceWidget/widgets/AudioToAudioWidget/AudioToAudioWidget.svelte:112]:
```suggestion
		for (const item of body) {
```

**Comment by julien-c** [comment]
Looks good from quick glance, but tagging @jonashaag and @mpariente with whom we implemented the audio-source-separation widget, to get their feedback

File: interfaces/Types.ts (hunk 3 above)
Comment (Narsil) [reply interfaces/Types.ts:51]: Yes !

File: widgets/src/lib/InferenceWidget/widgets/AudioToAudioWidget/AudioToAudioWidget.svelte (hunk 4 above)
Comment (Narsil) [reply widgets/src/lib/InferenceWidget/widgets/AudioToAudioWidget/AudioToAudioWidget.svelte:112]: Thanks

**Review by julien-c — APPROVED** [review]

**Comment by jonashaag** [comment]
Thanks for tagging me, I don't have anything to say to this PR but added comments on a related one

**Pull request merged**